api/lib/source/polkadotV0-source.js

```diff
@@ -213,16 +213,88 @@ class polkadotAPI {
         validator.votingPower = 0
       }
     })
-
-    return allValidators.map((validator) =>
+    const validatorsWithoutProfiles = allValidators.map((validator) =>
       this.reducers.validatorReducer(this.network, validator)
     )
+    if (profile) {
+      return { validators: allValidators, validatorsWithoutProfiles }
+    } else {
+      return validatorsWithoutProfiles
+    }
+  }
+
+  async getProfilesForValidators(
```

Comment:
all this code is duplicated and should be refactored

Reply:
?

Reply:
I don't see any duplication. Can you clarify? Such short comments don't help to properly review. At least they don't help me understand what you mean